Zend/zend_API.c

```diff
@@ -2015,11 +2027,15 @@ ZEND_API void zend_check_magic_method_implementation(const zend_class_entry *ce,
 			zend_error(error_type, "Method %s::__callStatic() must take exactly 2 arguments", ZSTR_VAL(ce->name));
 		} else if (QUICK_ARG_SHOULD_BE_SENT_BY_REF(fptr, 1) || QUICK_ARG_SHOULD_BE_SENT_BY_REF(fptr, 2)) {
 			zend_error(error_type, "Method %s::__callStatic() cannot take arguments by reference", ZSTR_VAL(ce->name));
+		}  else if (ZEND_TYPE_IS_SET(fptr->common.arg_info[0].type) && ZEND_TYPE_PURE_MASK(fptr->common.arg_info[0].type) != MAY_BE_STRING) {
+			zend_error(error_type, "Type of first argument of %s::__callStatic() must be string", ZSTR_VAL(ce->name));
+		} else if (ZEND_TYPE_IS_SET(fptr->common.arg_info[1].type) && ZEND_TYPE_PURE_MASK(fptr->common.arg_info[1].type) != MAY_BE_ARRAY) {
+			zend_error(error_type, "Type of second argument of %s::__callStatic() must be array", ZSTR_VAL(ce->name));
 		}
  	} else if (name_len == sizeof(ZEND_TOSTRING_FUNC_NAME) - 1 &&
  		!memcmp(lcname, ZEND_TOSTRING_FUNC_NAME, sizeof(ZEND_TOSTRING_FUNC_NAME)-1) && fptr->common.num_args != 0
 	) {
-		zend_error(error_type, "Method %s::%s() cannot take arguments", ZSTR_VAL(ce->name), ZEND_TOSTRING_FUNC_NAME);
+		zend_error(error_type, "Method %s::__toString() cannot take arguments", ZSTR_VAL(ce->name));
 	} else if (name_len == sizeof(ZEND_DEBUGINFO_FUNC_NAME) - 1 &&
 		!memcmp(lcname, ZEND_DEBUGINFO_FUNC_NAME, sizeof(ZEND_DEBUGINFO_FUNC_NAME)-1) && fptr->common.num_args != 0) {
 		zend_error(error_type, "Method %s::%s() cannot take arguments", ZSTR_VAL(ce->name), ZEND_DEBUGINFO_FUNC_NAME);
```

Comment:
```suggestion
		zend_error(error_type, "Method %s::__debugInfo() cannot take arguments", ZSTR_VAL(ce->name));
```
like you did for `__toString` above (and was done for `__callStatic` in #1049)? or maybe better to extract in a separate PR?

Reply:
A good idea! I will extract that into its own PR.

Reply:
Done #5398.